namesys/routing.go

```diff
@@ -19,25 +22,55 @@ var log = logging.Logger("namesys")
 // routingResolver implements NSResolver for the main IPFS SFS-like naming
 type routingResolver struct {
 	routing routing.IpfsRouting
+
+	cache     map[string]cacheEntry
+	cachelock sync.Mutex
+	cachelife time.Duration
 }
 
-// NewRoutingResolver constructs a name resolver using the IPFS Routing system
-// to implement SFS-like naming on top.
-func NewRoutingResolver(route routing.IpfsRouting) Resolver {
-	if route == nil {
-		panic("attempt to create resolver with nil routing system")
+func (r *routingResolver) cacheGet(name string) (path.Path, bool) {
+	r.cachelock.Lock()
+	entry, ok := r.cache[name]
+	r.cachelock.Unlock()
+	if ok && time.Now().Before(entry.eol) {
+		return entry.val, true
 	}
 
-	return &routingResolver{routing: route}
+	return "", false
 }
 
-// newRoutingResolver returns a resolver instead of a Resolver.
-func newRoutingResolver(route routing.IpfsRouting) resolver {
+func (r *routingResolver) cacheSet(name string, val path.Path, rec *pb.IpnsEntry) {
+	cacheTil := time.Now().Add(r.cachelife)
+	eol, ok := checkEOL(rec)
+	if ok && eol.Before(cacheTil) {
+		cacheTil = eol
```

Comment:
this is an interesting case of go's unused variable error. Its used, then assigned to, then that assigned value is never used.